paasta_tools/api/tweens/request_logger.py

```
@@ -0,0 +1,113 @@
+        )
+
+
+class request_logger_tween_factory:
+    """Tween that logs information about requests"""
+
+    def __init__(self, handler, registry):
+        self.handler = handler
+        self.registry = registry
+        self.log_name = registry.settings.get(
+            "request_log_name", DEFAULT_REQUEST_LOG_NAME,
+        )
+
+    def _log(
+        self, endpoint, level=logging.INFO, additional_fields=None,
+    ):
+        if clog is not None:
+            dct = {
+                "timestamp": time.time(),
+                "hostname": utils.get_hostname(),
+                "level": logging.getLevelName(level),
+                "cluster": settings.cluster,
+                "endpoint": endpoint,
+            }
+            if additional_fields is not None:
+                dct.update(additional_fields)
+            line = json.dumps(dct, sort_keys=True)
+            clog.log_line(self.log_name, line)
+
+    def __call__(self, request):
+        start_time = time.time()  # start clock for response time
+        request_fields = {
+            "path": request.path,
+            "params": request.params.mixed(),
+        }
+        response_fields = {}
+        log_level = logging.INFO
+
+        try:
+            response = self.handler(request)
+
+            response_fields["status_code"] = response.status_int
+            if 300 <= response.status_int < 400:
+                log_level = logging.WARNING
+            elif 400 <= response.status_int < 600:
+                log_level = logging.ERROR
+                response_fields["body"] = response.body.decode("utf-8")
+
+            return response
+
+        except Exception as e:
+            log_level = logging.ERROR
+            response_fields.update(
+                {
+                    "status_code": 500,
+                    "exc_type": type(e).__name__,
+                    "exc_info": traceback.format_exc(),
+                }
+            )
+            raise
```

why `raise` instead of `return` here if we've updated response fields?

---

disregard that, i misread what response_fields are